app/controllers/admin/orders_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Admin::OrdersController < Admin::AdminController
+  before_action :find_order_by_id, only: :update
+
+  def index
+    @pagy_orders, @orders = if filter_orders_by_status_text?
+                              filter_orders_by_status_text
+                            elsif params[:search_text].present?
+                              filter_orders_by_text
+                            elsif filter_orders_by_status?
+                              filter_orders_by_status
+                            else
+                              pagy Order.newest
+                            end
+  end
+
+  def update
+    @order.update status: params[:status]
+    @pagy_orders, @orders = pagy Order.newest
+    respond_to do |format|
+      format.html{redirect_to admin_orders_path}
+      format.js
+    end
+  end
+
+  private
+  def find_order_by_id
+    @order = Order.find_by id: params[:id]
+    return if @order
```

Comment:
Phải nghiên cứu cách nào tối ưu hơn chứ em. Ở đây em làm có 2 cái tìm kiếm mà phải viết ra một nấy các func thì em xem những dự án thực tế như tiki, thegioididong,... cái filer của nó sẽ to đến mức nào